src/v/archival/ntp_archiver_service.cc

```diff
@@ -508,6 +508,48 @@ ss::future<> ntp_archiver::sync_manifest_until_term_change() {
     }
 }
 
+ss::future<> ntp_archiver::unsafe_reset_metadata(ss::input_stream<char> is) {
+    // Deserialize outside of lock.
```

Comment:
I don't see a lock below?

I'm wondering how safe this is if there are concurrent uploads and/or upload_housekeeping_service stuff going on.  There is a pause mechanism for uploads that is used during leadership transfer, and housekeeping tasks can be de-registered from the housekeeping service, if we want to ensure that nothing is going on while we do this.

Reply:
My bad, I removed an explicit acquisition of the manifest lock here because it gets taken during apply. The point of the lock is to make decisions/access to the manifest atomic with respect to one another, and since there's no real business logic in resetting the manifest (e.g. finding a segment based on the manifest, uploading, etc, as in the archiver loop), we needn't hold it longer than the duration of the apply of the records.